src/Psl/Math/base_convert.php

```diff
@@ -21,20 +21,15 @@
  * done to arbitrary precision.
  *
  * @param non-empty-string $value
+ * @param int<2, 36> $from_base
+ * @param int<2, 36> $to_base
  *
  * @pure
  *
- * @throws Psl\Exception\InvariantViolationException If $value is empty, $from_base and $to_base are
- *                                                   out of the [2, 36] range, or the given value is invalid.
+ * @throws Psl\Exception\InvariantViolationException If the given value is invalid.
  */
 function base_convert(string $value, int $from_base, int $to_base): string
 {
-    Psl\invariant('' !== $value, 'Unexpected empty string, expected number in base %d', $from_base);
-    Psl\invariant(
-        $from_base >= 2 && $from_base <= 36,
-        'Expected $from_base to be between 2 and 36, got %d',
-        $from_base
-    );
     Psl\invariant($to_base >= 2 && $to_base <= 36, 'Expected $to_base to be between 2 and 36, got %d', $to_base);
```

Comment:
i left this on purpose to see if psalm would complain about it, but apparently it can't detect that this is redundant.